index.js

```diff
@@ -145,83 +190,39 @@ Toolkit.run(
     );
 
-    if (!content.length) {
-      tools.exit.success(
+    if (!events.length) {
+      return tools.exit.success(
         "No PullRequest/Issue/IssueComment/Release events found. Leaving README unchanged with previous activity"
       );
     }
 
-    if (content.length < 5) {
-      tools.log.info("Found less than 5 activities");
+    // Remove all lines between <!--START_SECTION:activity--> and <!--END_SECTION:activity--> comment
+    if (endIdx !== -1) {
+      readmeContent.splice(startIdx + 1, endIdx - startIdx);
     }
 
-    if (startIdx !== -1 && endIdx === -1) {
-      // Add one since the content needs to be inserted just after the initial comment
-      startIdx++;
-      content.forEach((line, idx) =>
-        readmeContent.splice(startIdx + idx, 0, `${idx + 1}. ${line}`)
+    if (events.length < MAX_LINES) {
+      tools.log.info(
+        `Found ${events.length} activities. Which is less than ${MAX_LINES} specified by MAX_LINES.`
       );
-
-      // Append <!--END_SECTION:activity--> comment
-      readmeContent.splice(
-        startIdx + content.length,
-        0,
-        "<!--END_SECTION:activity-->"
-      );
-
-      // Update README
-      fs.writeFileSync(`./${TARGET_FILE}`, readmeContent.join("\n"));
-
-      // Commit to the remote repository
-      try {
-        await commitFile();
-      } catch (err) {
-        tools.log.debug("Something went wrong");
-        return tools.exit.failure(err);
-      }
-      tools.exit.success("Wrote to README");
     }
 
-    const oldContent = readmeContent.slice(startIdx + 1, endIdx).join("\n");
-    const newContent = content
-      .map((line, idx) => `${idx + 1}. ${line}`)
-      .join("\n");
-
-    if (oldContent.trim() === newContent.trim())
-      tools.exit.success("No changes detected");
-
     startIdx++;
 
-    // Recent GitHub Activity content between the comments
-    const readmeActivitySection = readmeContent.slice(startIdx, endIdx);
-    if (!readmeActivitySection.length) {
-      content.some((line, idx) => {
-        // User doesn't have 5 public events
-        if (!line) {
-          return true;
-        }
-        readmeContent.splice(startIdx + idx, 0, `${idx + 1}. ${line}`);
-      });
-      tools.log.success(`Wrote to ${TARGET_FILE}`);
-    } else {
-      // It is likely that a newline is inserted after the <!--START_SECTION:activity--> comment (code formatter)
-      let count = 0;
-
-      readmeActivitySection.some((line, idx) => {
-        // User doesn't have 5 public events
-        if (!content[count]) {
-          return true;
-        }
-        if (line !== "") {
-          readmeContent[startIdx + idx] = `${count + 1}. ${content[count]}`;
-          count++;
-        }
-      });
-      tools.log.success(`Updated ${TARGET_FILE} with the recent activity`);
-    }
+    // Append new content
+    events.forEach((line, idx) =>
+      readmeContent.splice(startIdx + idx, 0, `${idx + 1}. ${line}`)
```

Comment:
```suggestion
      readmeContent.splice(startIdx + idx, 1, `${idx + 1}. ${line}`)
```

There can be an edge case where the `<!--END_SECTION:activity-->` comment got removed, and the further runs will append entries towards the end rather than replacing them.

Reply:
On second thought, how about replacing the entry altogether instead of the `splice` approach?

```suggestion
      readmeContent[startsIdx + idx] = `${idx + 1}. ${line}`
```

Reply:
On second thought, how about replacing the entry altogether instead of the `splice` approach?

```suggestion
      readmeContent[startIdx + idx] = `${idx + 1}. ${line}`
```

Reply:
I think you might have misunderstood how the logic works.

If the end section comment has been found all lines between Start and including the end section comment will be removed:
https://github.com/jamesgeorge007/github-activity-readme/blob/rc/0.5.0/index.js#L202-L205

The `forEach` in line in line `215 - 217` uses the `splice` method (starting after the start section comment) to insert each single message after each other. We cannot use the array notation here as that might cause an array out of bounds exception or overwrite any following other lines.

After each message was inserted using `splice` we add back the end section comment:

https://github.com/jamesgeorge007/github-activity-readme/blob/rc/0.5.0/index.js#L220-L225

Reply:
But you are right, if the end comment got removed by the user by accident and the old messages do still exist. We would append those messages with the new ones and put the new end comment in between. That would cause the issue that we would keep the old message forever. Unfortunately, I don't see any easy solution to deal with that kind of user error. How would we identify what is part of our list and what was added by the user on purpose? Do you have any ideas?

Reply:
Yeah, let's not handle the case if the `<!--END_SECTION:activity-->` comment got removed later on; was wondering if it would be helpful to log a message in such a case.

> If there're empty lines after the <!--START_SECTION:activity--> comment. This is handled if the <!--END_SECTION:activity--> comment is present. If it's not the case and if the line after the <!--START_SECTION:activity--> comment is empty, let's show a message conveying the <!--END_SECTION:activity--> comment was not found, and it will be inserted towards the end. Also, empty lines were found after the <!--START_SECTION:activity--> comment and will be replaced by the activity events.